cmd/odf/set/log_level.go

```diff
@@ -6,44 +6,33 @@ import (
 
 	"github.com/red-hat-storage/odf-cli/cmd/odf/root"
 	"github.com/red-hat-storage/odf-cli/pkg/rook/common"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
 	"github.com/spf13/cobra"
 )
 
 var setLogLevelCmd = &cobra.Command{
 	Use:                "log-level",
-	Short:              "Set different log levels for ceph dameons like mon, osd and mds",
-	Example:            "odf set log-level osd 20",
+	Short:              "Set different log levels for ceph components like mon, osd and mds",
+	Example:            "odf set log-level osd crush 10",
 	DisableFlagParsing: true,
-	PreRunE:            validateLogLevelArgs,
-	Args:               cobra.ExactArgs(2),
+	Args:               cobra.ExactArgs(3),
 	Run: func(cmd *cobra.Command, args []string) {
+		validateLogLevelArgs(args[2])
 		ctx := cmd.Context()
 		clientsets := root.GetClientsets(ctx)
-		common.SetLogLevel(ctx, clientsets, root.OperatorNamespace, root.StorageClusterNamespace, args)
+		common.SetLogLevel(ctx, clientsets, root.OperatorNamespace, root.StorageClusterNamespace, args[0], args[1], args[2])
 	},
 }
 
-func validateLogLevelArgs(_ *cobra.Command, args []string) error {
-	isValidSubsytem := false
-	for _, subsystem := range common.ValidCephSubSystems {
-		if subsystem == args[0] {
-			isValidSubsytem = true
-			break
-		}
-	}
-
-	if !isValidSubsytem {
-		return fmt.Errorf("invalid ceph subsystem %q. Supported subsystems : %+v", args[0], common.ValidCephSubSystems)
-	}
-
-	// validate log level
-	match, err := regexp.MatchString("^[1-9][0-9]?$", args[1])
+func validateLogLevelArgs(logLevel string) error {
+	// validate {log-level}/{memory-level} format. Examples of vaild formats: 5, 1/20, 10/20, etc.
+	match, err := regexp.MatchString("^[1-9][0-9]?(\\/[1-9][0-9]?)?$", logLevel)
```

Comment:
Does ceph fail if an invalid log level is set? If so, we don't even need to validate this.

Reply:
Yes. Ceph does is good job in validating log levels.

``` Use "odf set [command] --help" for more information about a command.
❯ bin/odf set ceph-log-level osd.0 test 34343
Error: command terminated with exit code 34
Error ERANGE: error parsing value: value must be in range [0, 99]
❯ bin/odf set ceph-log-level osd.0 test h
Error: command terminated with exit code 22
Error EINVAL: error parsing value: value must take the form N or N/M, where N and M are integers
❯ bin/odf set ceph-log-level osd.0 test h/7
Error: command terminated with exit code 22
Error EINVAL: error parsing value: value must take the form N or N/M, where N and M are integers
❯ bin/odf set ceph-log-level osd.0 test h/k
Error: command terminated with exit code 22
Error EINVAL: error parsing value: value must take the form N or N/M, where N and M are integers

❯ bin/odf set ceph-log-level osd.0 test 4jhj

```
Although the last one is invalid, it got passed. But ceph has set the log/memory level to `4/4` for the last command. So we are good to remove this validation as well. 

Now we don't have any validation for this odf command args. Since we are adding the `debug_` prefix to ceph subsystem, I don't user can use this command just to set any config option (which was the initial concern)